src/main/java/TaskList.java

```diff
@@ -0,0 +1,157 @@
+            for (String i : listHolder) {
+                date = date + " " + commandList.remove(0);
+            }
+
+            Date dateHolder;
+            try {
+                dateHolder = DATE_FORMAT.parse(date);
+            } catch (ParseException e) {
+                throw new IODukeException("Please enter date in this format: d/m/y HHmm");
+            }
+
+            Task taskHolder = new Deadline(stringHolder, dateHolder, false);
+            taskList.add(taskHolder);
+
+            String holder = "Got it. I've added this task:\n" + "  [D][\u2718] " + stringHolder + " (by: " + date + ")" + "Now you have " + taskList.size() + " tasks in the list.";
+            return holder;
+        } else {
+            stringHolder = commandList.remove(0);
+            listHolder.remove(0);
+            for (String i : listHolder) {
+                if (i.equals("/at")) {
+                    commandList.remove(0);
+                    break;
+                } else {
+                    stringHolder = stringHolder + " " + commandList.remove(0);
+                }
+            }
+
+            String date = commandList.remove(0);
+            listHolder.clear();
+            listHolder.addAll(commandList);
+            for (String i : listHolder) {
+                date = date + " " + commandList.remove(0);
+            }
+
+            Date dateHolder;
+            try {
+                dateHolder = DATE_FORMAT.parse(date);
+            } catch (ParseException e) {
+                throw new IODukeException("Please enter date in this format: 2/12/2019 1800");
+            }
+
+            Task taskHolder = new Event(stringHolder, dateHolder, false);
+            taskList.add(taskHolder);
+
+            String holder = "Got it. I've added this task:\n" + "  [E][\u2718] " + stringHolder + " (at: " + date + ")" + "Now you have " + taskList.size() + " tasks in the list.";
+            return holder;
+        }
+    }
+
+    /**
+     * Changes the status of the task from undone to done
+     * @param index Index of the task
+     */
+    public String doneTask(int index) {
+        index = index - 1;
+        taskList.get(index).changeStatusTrue();
+        String holder = ui.horizontalLine() + "\nNice! I've marked this task as done:\n" + "  [" + taskList.get(index).getType() + "]" + "[" + taskList.get(index).getStatusIcon() + "] " + taskList.get(index).getDescription() + taskList.get(index).getDate() + "\n" + ui.horizontalLine() + "\n";
+        return holder;
+    }
+
+    /**
+     * Deletes the task from the list
+     * @param index Index of the task
+     */
+    public String deleteTask(int index) {
+        index = index - 1;
+        Task taskHolder = taskList.remove(index);
+        String holder = ui.horizontalLine() + "\nNoted. I've removed this task:\n" + "  [" + taskHolder.getType() + "]" + "[" + taskHolder.getStatusIcon() + "] " + taskHolder.getDescription() + taskHolder.getDate() + "\n" + ui.horizontalLine() + "\n";
```

Comment:
If the String becomes too complicated, you can try looking at StringBuilder. Aside from making the code looks simpler, StringBuilder is usually faster than String concatenation operation. From what I know, String is an Immutable object and therefore every concatenation operation will result in a new String Object.

Reply:
Noted, thank you for the tip provided!